trieste/acquisition/rule.py

```diff
@@ -1624,7 +1626,13 @@ def __init__(
         self._y_min = tf.constant(np.inf, dtype=self.location.dtype)
 
     def _init_eps(self) -> None:
-        self.eps = self._zeta * (self.global_search_space.upper - self.global_search_space.lower)
+        if not isinstance(self.global_search_space, HasOneHotEncoder):
```

Comment:
Using an `isinstance` check in this file to detect categorical spaces makes me a bit uncomfortable, as it forces developers to have to inherit from `HasOneHotEncoder`, and it feels too much of a special case. I think our original design decision was to use a property like `is_categorical` to do this. I don't have a strong objection, but just wanted to highlight that. I think we already discussed this before, but can't remeber the conclusion.

Reply:
Remind me: are there any categorical spaces (where we would wish to use Hamming distances) that are also numerically bounded? Because looking at it now, it seems much more natural to write

```suggestion
        if self.global_search_space.has_bounds
```

Certainly we shouldn't calculate eps for unbounded spaces.

Reply:
I don't think so. We won't have categorical spaces that have bounds. However, we could potentially have the reverse, i.e. spaces that are not bounded and are also not categorical.

Reply:
Well we can't use eps as written for unbounded spaces, even if they're not categorical, as it uses the bounds. Is there any reason not to default to Hamming distance for those cases, at least for now?

Reply:
That seems fine to me. @vpicheny what do you think?